github/repos_rules.go

```diff
@@ -0,0 +1,392 @@
+	case "required_deployments":
+		RepositoryRule.Parameters = &RequiredDeploymentEnvironmentsRuleParameters{}
+		if err := json.Unmarshal(data, &RepositoryRule); err != nil {
+			return err
+		}
+		r.Parameters = RepositoryRule.Parameters
+	case "commit_message_pattern", "commit_author_email_pattern", "committer_email_pattern", "branch_name_pattern", "tag_name_pattern":
+		RepositoryRule.Parameters = &RulePatternParameters{}
+		if err := json.Unmarshal(data, &RepositoryRule); err != nil {
+			return err
+		}
+		r.Parameters = RepositoryRule.Parameters
+	case "pull_request":
+		RepositoryRule.Parameters = &PullRequestRuleParameters{}
+		if err := json.Unmarshal(data, &RepositoryRule); err != nil {
+			return err
+		}
+		r.Parameters = RepositoryRule.Parameters
+	case "required_status_checks":
+		RepositoryRule.Parameters = &RequiredStatusChecksRuleParameters{}
+		if err := json.Unmarshal(data, &RepositoryRule); err != nil {
+			return err
+		}
+		r.Parameters = RepositoryRule.Parameters
+	default:
+		r.Type = ""
+		r.Parameters = nil
+		return fmt.Errorf("RepositoryRule.Type %T is not yet implemented, unable to unmarshal", RepositoryRule.Type)
+	}
+
+	return nil
+}
+
+// NewCreationRule creates a rule to only allow users with bypass permission to create matching refs.
+func NewCreationRule() (rule RepositoryRule) {
+	return RepositoryRule{
+		Type: "creation",
+	}
+}
+
+// NewUpdateRule creates a rule to only allow users with bypass permission to update matching refs.
+func NewUpdateRule(params *UpdateAllowsFetchAndMergeRuleParameters) (rule RepositoryRule) {
+	return RepositoryRule{
+		Type:       "update",
+		Parameters: params,
+	}
+}
+
+// NewDeletionRule creates a rule to only allow users with bypass permissions to delete matching refs.
+func NewDeletionRule() (rule RepositoryRule) {
+	return RepositoryRule{
+		Type: "deletion",
+	}
+}
+
+// NewRequiredLinearHistoryRule creates a rule to prevent merge commits from being pushed to matching branches.
+func NewRequiredLinearHistoryRule() (rule RepositoryRule) {
+	return RepositoryRule{
+		Type: "required_linear_history",
+	}
+}
+
+// NewRequiredDeploymentsRule creates a rule to require environments to be successfully deployed before they can be merged into the matching branches.
+func NewRequiredDeploymentsRule(params *RequiredDeploymentEnvironmentsRuleParameters) (rule RepositoryRule) {
+	return RepositoryRule{
+		Type:       "required_deployments",
+		Parameters: params,
+	}
+}
+
+// NewRequiredSignaturesRule creates a rule a to require commits pushed to matching branches to have verified signatures.
+func NewRequiredSignaturesRule() (rule RepositoryRule) {
+	return RepositoryRule{
+		Type: "required_signatures",
+	}
+}
+
+// NewPullRequestRule creates a rule to require all commits be made to a non-target branch and submitted via a pull request before they can be merged.
+func NewPullRequestRule(params *PullRequestRuleParameters) (
+	rule RepositoryRule) {
```

Comment:
There's no need to break this line here... please add the return value here to the end of line 195.